src/users/users.service.ts

```diff
@@ -26,8 +33,30 @@ export class UsersService {
     return this.userRepository.save(user);
   }
 
+  async createTokenForUser(
+    userName: string,
+    identifier: string,
+  ): Promise<AuthToken> {
+    const user = await this.getUserByUsername(userName);
+    let accessToken = '';
+    for (let i = 0; i < 100; i++) {
+      try {
+        const randomString = crypt.randomBytes(64).toString();
+        accessToken = await this.hashPassword(randomString);
+        await this.getUserByAuthToken(accessToken);
+      } catch (NotInDBError) {
+        const token = AuthToken.create(user, identifier, accessToken);
+        return this.authTokenRepository.save(token);
```

Comment:
The random string has to be returned and send to the user which can be hashed if it is send back to the user. After that the server does not know the random string anymore only the hash of it.

Reply:
done